### ISSUE TITLE
Performance on cityscapes

### ISSUE THREAD

**Issue by EchoAmor** [YudeWang/deeplabv3plus-pytorch#5]
Thanks for your work!I am a student .
I have trained your model on Cityscapes , I only modified  train.py  and  config.py in the directory /deeplabv3plus-pytorch/experiment/deeplabv3+voc/  and  dataset's path in cityscapes.py in  /datasets/,and used 2 gpus.But the results is very bad ,test results only get 19.89% ,and I see images during training  by tensorboardX, the images seems lose some classes.

Did I do something wrong ?Is there anything I haven't modified? Hope for your respond! Thanks very much!

![1](https://user-images.githubusercontent.com/25685837/51481210-4afbdf80-1dce-11e9-8a7e-967242af5b91.png)
![2](https://user-images.githubusercontent.com/25685837/51481219-4e8f6680-1dce-11e9-943d-9a2a87e3afff.png)
![3](https://user-images.githubusercontent.com/25685837/51481221-4f27fd00-1dce-11e9-95cd-81e04847f78e.png)

**Comment by YudeWang** [comment]
I see the training step in tensorboardX is only 8500 which is not enough. I set 46 epochs for PASCAL VOC 2012 which equal to about 30k iterations. For Cityscapes dataset, my experiment set epochs=160 with batchsize=8. You can change the setting according to your devices but please keep enough training iteration. 

In deeplabv3+ origin paper, it has said that the global average pooling branch of ASPP will hurt the performance on cityscapes, please remove it.

The repository has been update recently, please update to the new version. Thanks for your comment!

**Comment by EchoAmor** [comment]
Thanks!I 'll try it again today.

**Comment by EchoAmor** [comment]
![407](https://user-images.githubusercontent.com/25685837/51660307-4874d200-1fe8-11e9-8489-bd505b8b31f1.png)
![406](https://user-images.githubusercontent.com/25685837/51660312-4dd21c80-1fe8-11e9-9660-6d91f3b58eba.png)
![405](https://user-images.githubusercontent.com/25685837/51660314-51fe3a00-1fe8-11e9-8a1d-1639b88b963d.png)
![404](https://user-images.githubusercontent.com/25685837/51660318-562a5780-1fe8-11e9-8c72-6d5f3fcd62b8.png)
![403](https://user-images.githubusercontent.com/25685837/51660326-59bdde80-1fe8-11e9-92c1-f4e940d5a94c.png)
![402](https://user-images.githubusercontent.com/25685837/51660330-5d516580-1fe8-11e9-972d-7f15c15e5f08.png)
![401](https://user-images.githubusercontent.com/25685837/51660335-617d8300-1fe8-11e9-98f3-2055feaf972f.png)

Hello! I have modified to 200 epoches and updated your work .Still 2 GPUs. But I didn't see the  global average pooling branch of ASPP,the results are still very low. Did you mean I need to remove ASPP? I don't know how to solve this problem.Have you trained on cityscapes? How about your results?

**Comment by Arenops** [comment]
Trained on cityscapes with 4 GPUs and batches==20  and lr==0.0006 in config.py in deeplabv3+voc, get **mIOU 24.867% after 32 epochs**. Loss is remain 0.3-0.6 and no longer reduce. 
@YudeWang any advices? Thanks. Still training and will update once get better results.

**Comment by YudeWang** [comment]
@EchoAmor Global average pooling branch is branch5 in ASPP.py. My experiment was did three month ago achieving 75% mIoU, and I believe new model can have better performance. I will retest it during winter vacation.

@Arenops Your initial learning rate is too small (I use 0.007) and iteration is not enough (>30k). Please check recommendation above.

**Comment by EchoAmor** [comment]
Thanks for your patience very much! I'll keep trying and update my results. @YudeWang

**Comment by EchoAmor** [comment]
@Arenops  Hello！ Have u got results now? Can u share your results? Thanks a lot !

**Comment by Flames60** [comment]
> @EchoAmor Global average pooling branch is branch5 in ASPP.py. My experiment was did three month ago achieving 75% mIoU, and I believe new model can have better performance. I will retest it during winter vacation.
> 
> @Arenops Your initial learning rate is too small (I use 0.007) and iteration is not enough (>30k). Please check recommendation above.

Hello, I found that the 

> @Arenops Hello！ Have u got results now? Can u share your results? Thanks a lot !

Hello, how about your performance? What is your lr, cropsize and pretrain-model?

**Comment by Flames60** [comment]
> @Arenops Hello！ Have u got results now? Can u share your results? Thanks a lot !

Hello, how about your performance? What is your lr, cropsize and pretrain-model?
